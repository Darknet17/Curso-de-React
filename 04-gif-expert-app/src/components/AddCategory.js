import {useState} from "react";
import PropTypes from 'prop-types';
import {Button, Form, FormControl} from "react-bootstrap";

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if(inputValue.trim().length > 2){
            setCategories(cat=>[inputValue, ...cat])
            setInputValue('');
        }

    }

    return (
        <Form className="d-flex" onSubmit={ handleSubmit }>
            <FormControl
                type="search"
                placeholder="Search"
                className="me-2"
                aria-label="Search"
                value={ inputValue }
                onChange={ handleInputChange }
            />
            <Button type="submit" variant="outline-success">Search</Button>
        </Form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
}

