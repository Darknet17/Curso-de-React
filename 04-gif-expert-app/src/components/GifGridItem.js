import PropTypes from "prop-types";
import {Card} from "react-bootstrap";

export const GifGridItem = ({ title, url }) => {
    return (
        <Card className="card animate__animated animate__fadeIn" style={{ width: '18rem' }}>
            <Card.Img variant="top" src={ url } />
            <Card.Body>
                <Card.Text> {
                    title
                }</Card.Text>
            </Card.Body>
        </Card>
    );
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url:  PropTypes.string.isRequired
}
