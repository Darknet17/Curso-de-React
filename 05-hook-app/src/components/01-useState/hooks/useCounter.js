import {useState} from "react";

export const useCounter = (initialState = 10 ) => {
    const [state, setState]  = useState(initialState); // 10

    const increment = (factor = 1) => {
        setState( state + factor);
    }

    const descrement = (factor = 1) => {
        setState( state - factor);
    }

    const reset = () => {
        setState(initialState);
    }

    return {
        state,
        increment,
        descrement,
        reset
    };

}
