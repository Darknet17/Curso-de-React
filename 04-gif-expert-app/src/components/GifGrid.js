import PropTypes from "prop-types";
import {GifGridItem} from "./GifGridItem";
import {useFetchGifs} from "../hooks/useFetchGifs";
import {Col, Row, Spinner} from "react-bootstrap";

export const GifGrid = ({ category }) => {

    const {data: images, loading} = useFetchGifs(category);

    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>

            {  loading && <Spinner animation="grow" variant="dark" />  }

            <Row>
                {
                    images.map(  ({id, title, url }) => (
                        <Col sm={3}>
                            <GifGridItem key={id} title={title} url={url}/>
                        </Col>
                    ))
                }
            </Row>
        </>
    );
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}


