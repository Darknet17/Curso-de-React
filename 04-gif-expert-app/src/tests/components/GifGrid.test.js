import {shallow} from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import {useFetchGifs} from "../../hooks/useFetchGifs";
// esto va a simular o responder la informacion
jest.mock("../../hooks/useFetchGifs");

describe('Probando en el <GifGrid/>', ()=> {

    const category = "One Punch";

    test('debe de mostrar el componente', ()=>{
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imagenes useFetchGifs', ()=>{

        const gifs = [{
            id: "23",
            url: "https://localhost:8080/cualquiercosa.jpg",
            title: "Cualquier cosa"
        },
            {
                id: "24",
                url: "https://localhost:8080/cualquiercosa.jpg",
                title: "Cualquier cosa"
            }
        ]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find("p").exists() ).toBe(false)
        expect( wrapper.find("GifGridItem").length ).toBe(gifs.length)
    });
});
