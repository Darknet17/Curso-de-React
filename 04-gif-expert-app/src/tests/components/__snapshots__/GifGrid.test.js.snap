// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando en el <GifGrid/> debe de mostrar el componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <Spinner
    animation="grow"
    variant="dark"
  />
  <Row />
</Fragment>
`;

exports[`Probando en el <GifGrid/> debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <Row>
    <Col
      key="23"
      sm={3}
    >
      <GifGridItem
        title="Cualquier cosa"
        url="https://localhost:8080/cualquiercosa.jpg"
      />
    </Col>
    <Col
      key="24"
      sm={3}
    >
      <GifGridItem
        title="Cualquier cosa"
        url="https://localhost:8080/cualquiercosa.jpg"
      />
    </Col>
  </Row>
</Fragment>
`;
